### ISSUE TITLE
Reimplement CLI in rust

### ISSUE THREAD

**Issue by kdy1** [swc-project/swc#1589]
Node binding should export `runCLI`.

This would help lots of cases.

**Comment by tmpfs** [comment]
Hey @kdy1, I keep looking at the Typescript CLI entry points and wonder why they exist and it's not just straight Rust binaries using `clap` (or `structopt`).

Is there functionality that is enabled by having the command line programs exposed as node executables? Is it part of a plan for a Javascript-based plugin system maybe?

I just wanted to look into the reason why `--ignore` isn't working for example and it seems like it was never implemented from what I can tell. Implementing this would be much more straightforward if it was all just Rust :pray:

**Comment by kdy1** [comment]
> Is there functionality that is enabled by having the command line programs exposed as node executables? Is it part of a plan for a Javascript-based plugin system maybe?

No. It's because to make managing version more convenient for the end-user

**Comment by tmpfs** [comment]
Thanks, that helps! I will try to find some time to tackle this one as I think it would simplify things quite a bit - we should be able to remove out quite a bit of typescript :+1: 

I assume you would prefer to use `clap` over `structopt`?

**Comment by kwonoj** [comment]
It's quite late to realize but I forget js-side plugin support in @swc/cli. This may be some headache if we want 100% feature parity.

For now I'll skip those until new cli becomes stable enough. This is also related with experimental plugin stablization and may possible to think about whole plan accordingly based on those.

**Comment by kwonoj** [comment]
`swc_cli` is a ground to implement this. Roughly, these are the tasks need to be implemented and if anyone want to dive in, please feel free to send PR

1. `compile` command
- port rest of cli flag features in subcommand flag struct
- `watch` mode implementation
2. `minify` command
3. `lint` command
4. `bundle` command - accept JSON based configs 

Currently, implementations are sort of exploration, does not have to be 1:1 match to existing `@swc/cli`'s behavior. Specifically, there will be rough edges to deal with dynamic values which requires js runtime. We want to give suffecient time to iterate, finalize behaviors based on feedbacks & some constraints we have.

**Comment by younes-io** [comment]
I would really like to work on one of the commands, but I need some guidance
Anyone could help me to start ?

**Comment by StemCll** [comment]
Hi @kwonoj 
Is this PR still valid? https://github.com/swc-project/swc/pull/5003

**Comment by stephanrotolante** [comment]
Hey @kwonoj,

Is this still a good first issue? I am looking back at some of the PRs referenced in the ticket and am trying to get an understand at whats going on. I am still kind of unsure whats being asked. 

If I had to guess its replace the entire [nodejs cli](https://github.com/swc-project/pkgs/tree/main/packages/cli) into thats a rust/node bind gen?
